### PR TITLE
Recipe popup ng

### LINKED ISSUE #30: Pantry - calculate cost for meal ingredients
For a user, calculate how much it will cost to buy the necessary ingredients needed to cook a given meal, based on what’s in the pantry.

### PR DESCRIPTION
**Pull request by nicolegooden**
## Description

This PR does the following:

+ allows user to click on a recipe card (targeting the recipe's image) to show the recipe info popup
+ allows user to click anywhere on the popup to remove it from view and return to the normal view of the page on load
+ allows user to see the recipe name, total cost of the recipe's ingredients, and instructions
**as of now, the user sees an array of objects which should be the recipe ingredients by name, so the `getRecipeIngredientNames()` function in `scripts.js` will need to be finished.  There is pseudocode in there for guidance!**

## Affected areas of application

Main affected areas include:

+ `domUpdates.js`
+ `scripts.js`
+ `index.html`

## How Has This Been Tested?

This has been tested by testing the UI, populating the recipes and clicking on the image of a single recipe card to show the popup, then clicking on the popup to hide it from view.

## Relevant Tickets

Closes #30  

## Screenshots

Screen Shot 2020-09-15 at 5 16 58 PM

## Comments

Definitely need to clean up this code!

### CODE REVIEW

File: src/domUpdates.js [hunk 1]
```diff
@@ -38,6 +38,30 @@ let domUpdates = {
        </article>`
       })
 
+  },
+
+  displayRecipeInfo(recipe, ingredients) {
```
Comment (taylorjohnson141) [inline src/domUpdates.js:43]: I think this is going to look really good!

File: src/domUpdates.js [hunk 2]
```diff
@@ -38,6 +38,30 @@ let domUpdates = {
...
+    <h1>${recipe.name}</h1>
+    <div class="recipe-ingredients-with-cost">
+      <p class="popup-ingredients">${JSON.stringify(recipe.ingredients)}</p>
+      <h3 class="pop-costs">${recipe.calculateCost()}</h3>
+    </div>
+    <p class="recipe-instructions">${trueInstructions}</p>
+</section>`
+  },
+
+  showRecipePopup() {
```
Comment (taylorjohnson141) [inline src/domUpdates.js:59]: We might be able to use toogle here instead!

File: test/domUpdates-test.js [hunk 3]
```diff
@@ -0,0 +1,5 @@
+import {expect} from 'chai';
```
Comment (taylorjohnson141) [inline test/domUpdates-test.js:1]: This is a good start!

**Review by taylorjohnson141 — APPROVED** [review]

File: src/index.html [hunk 4]
```diff
@@ -43,32 +43,18 @@ <h3 class="page-title">All Recipes</h3>
...
-          <article class="recipe-name-area">
-            <h2 class="recipe-name">Recipe Name</h2>
-          </article>
-        </article> -->
       </section>
     </main>
       <section class="recipe-display hidden">
         <section class="recipe-display-main">
-          <div class="recipe-display-left">
-            <div></div>
```
Comment (CalebCyphers) [inline src/index.html:63]: Thanks for getting rid of these divs

File: src/scripts.js [hunk 5]
```diff
@@ -72,8 +72,23 @@ if(event.target.classList.contains('star-icon')) {
...
 })
 
+function getRecipeIngredientNames() {
+  //access the ingredients from ingredientData
+  //compare each ingredient and see if the ingredient
+  //is included in the recipe.ingredients (by ID)
+  //if true => returns the ingredient that matches
+  //set those included ingredients to a variable
+  //for each of those found ingredients, reassign
+  //values by creating new keys
```
Comment (CalebCyphers) [inline src/scripts.js:89]: We need to clean this up before submission

File: src/scripts.js [hunk 6]
```diff
@@ -168,18 +191,17 @@ function grabUsers() {
 function mergeFetchTimelines() {
  Promise.all([grabUsers(), grabIngredients(), grabRecipes()])
  .then(values => {
-   let usersData = values[0]
-   ingredientsData = values[1]
+   let usersData = values[0];
+   ingredientsData = values[1];
```
Comment (CalebCyphers) [inline src/scripts.js:195]: Thanks for adding semicolons

**Review by CalebCyphers — APPROVED** [review]

**Pull request merged**